src/http/headers.cr

```diff
@@ -55,25 +64,77 @@ struct HTTP::Headers
     @hash = Hash(Key, String | Array(String)).new
   end
 
-  def []=(key, value : String)
+  # Sets the value of *key* header to *value*.
+  #
+  # Previous values are overriden.
+  # Raises `ArgumentError` if *value* includes invalid characters.
+  #
+  # ```
+  # require "http/headers"
+  #
+  # headers = HTTP::Headers{"host" => "crystal-lang.org"}
+  # headers["host"] = "play.crystal-lang.org"
+  # headers # => HTTP::Headers{"host" => "play.crystal-lang.org"}
+  # headers["accept-encoding"] = ["text/html", "application/json"]
+  # headers # => HTTP::Headers{"host" => "play.crystal-lang.org", "accept-encoding" => ["text/html", "application/json"]}
+  # ```
+  def []=(key : String, value : String) : String
     check_invalid_header_content(value)
 
     @hash[wrap(key)] = value
   end
 
-  def []=(key, value : Array(String))
+  # Sets the value of *key* header to *value*.
+  #
+  # Previous values are overriden.
+  # `ArgumentError` will be raised if header includes invalid characters.
+  #
+  # ```
+  # require "http/headers"
+  #
+  # headers = HTTP::Headers{"host" => "crystal-lang.org"}
+  # headers["host"] = ["crystal-lang.org", "play.crystal-lang.org"]
+  # headers # => HTTP::Headers{"host" => ["crystal-lang.org", "play.crystal-lang.org"]}
+  # headers["accept-encoding"] = ["text/html", "application/json"]
+  # headers # => HTTP::Headers{"host" => ["crystal-lang.org", "play.crystal-lang.org"], "accept-encoding" => ["text/html", "application/json"]}
+  # ```
+  def []=(key : String, value : Array(String)) : Array(String)
     value.each { |val| check_invalid_header_content val }
 
     @hash[wrap(key)] = value
   end
 
-  def [](key)
+  # Returns the value of *key* header.
+  #
+  # If there are multiple values, they are joind to a comma-separated string.
```

Comment:
_They are joined_ (with a typo)? That reads weird...

```suggestion
  # If there are multiple values, they are joined to a comma-separated string.
```

I'd suggest sth along the lines of:

```suggestion
  # In case of multiple values, they are returned as a comma-separated string.
```